Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3279,11 +3280,30 @@ void WebPageProxy::processNextQueuedMouseEvent()
 
     LOG_WITH_STREAM(MouseHandling, stream << "UIProcess: sent mouse event " << eventType << " (queue size " << internals().mouseEventQueue.size() << ")");
     m_process->recordUserGestureAuthorizationToken(event.authorizationToken());
-    sendWithAsyncReply(Messages::WebPage::MouseEvent(m_mainFrame->frameID(), event, sandboxExtensions), [this, protectedThis = Ref { *this }] (WebEventType eventType, bool handled) {
+    CompletionHandler<void(WebEventType, bool, std::optional<RemoteMouseEventData>)> completionHandler = [this, protectedThis = Ref { *this }, sandboxExtensions = WTFMove(sandboxExtensions), &completionHandler] (WebEventType eventType, bool handled, std::optional<WebCore::RemoteMouseEventData> remoteMouseEventData) {
         if (!m_pageClient)
             return;
+
+        if (remoteMouseEventData) {
+            auto* frame = WebFrameProxy::webFrame(remoteMouseEventData->targetFrameID);
+            if (!frame)
+                return;
+
+            auto remotePageProxy = frame->remotePageProxy();
+            if (!remotePageProxy)
+                return;
```

Comment:
This can happen if a.com has an iframe at b.com which has an iframe at a.com and you click on the grandchild iframe.  You'd get a WebFrameProxy that doesn't have a RemotePageProxy because its WebPage communicates directly with the WebPageProxy.  In this case, we'd want to propagate the event and sandbox extensions back to the WebPageProxy.